server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2177,8 +2177,7 @@ private boolean shouldPeriodicallyFlush(long flushThresholdSizeInBytes, long flu
 
     @Override
     protected void flushHoldingLock(boolean force, boolean waitIfOngoing, ActionListener<FlushResult> listener) throws EngineException {
-        assert isClosed.get() == false; // might be closing, but not closed yet
```

Comment:
Did you consider changing it to:

```
assert isClosed.get() == false || failedEngine.get() != null;
```
?

Reply:
I did, but it doesn't seem especially useful - it says we're either open, or closed due to a failure, but the reason for closing doesn't really matter here, AFAICT it'd be fine if we closed gracefully while running this method too.

Reply:
Makes sense, thanks for clarifying.